include/ebpf_extension.h

```diff
@@ -91,18 +94,44 @@ typedef struct _ebpf_extension_program_dispatch_table
     ebpf_program_batch_end_invoke_function_t ebpf_program_batch_end_invoke_function;
 } ebpf_extension_program_dispatch_table_t;
 
-typedef struct _ebpf_extension_data
+typedef struct _ebpf_extension_data_v0
 {
     ebpf_extension_header_t header;
     const void* data;
-} ebpf_extension_data_t;
+} ebpf_extension_data_v0_t;
+
+typedef struct _ebpf_extension_data_v1
+{
+    ebpf_extension_header_t header;
+    const void* data;
+    union
+    {
+        uint64_t as_uint64;
+        struct
+        {
+            bool prog_attach_flags : 1; ///< Program attach flags are supported.
```

Comment:
I think we should not use bit fields in versioned structs. Bit fields size can change over time (as new bits are added), which will break backward compat tests.

Reply:
It is part of a union with a uint64_t so as to reserve 64bits, permitting it to add up to 64bits to the bit field. I think it's ok to use a bit field and instead people need to be aware that adding if they need to add > 64 flags they will need to add a new bit field.

Reply:
Bit fields are fine for back compat as long as when new bits are used for some purpose, 0 means the old behavior.